chain/chain/src/resharding/event_type.rs

```diff
@@ -0,0 +1,184 @@
+//! Collection of all resharding V3 event types.
+
+use near_chain_primitives::Error;
+use near_primitives::hash::CryptoHash;
+use near_primitives::shard_layout::ShardLayout;
+use near_primitives::types::AccountId;
+use near_store::ShardUId;
+use tracing::error;
+
+/// Struct used to destructure a new shard layout definition into the resulting resharding event.
+#[derive(Debug, Clone)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub enum ReshardingEventType {
+    /// Split of a shard.
+    SplitShard(ReshardingSplitShardParams),
+}
```

Comment:
This should be moved to the shared resharding code right? As in it's not flat storage specific? Fine to do it separately.